From fd91c45b48ce62479f8e319a37c9f170cabc089f Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Tue, 5 Apr 2016 14:34:22 +0500
Subject: [PATCH] gst/vaapi: decrease priority of plugins

Depending upon the VA capabilities of a target there
can be several types of streams that are not playable
using gstreamer-vaapi mainly depending on profiles
etc.
We now decrease the priority of all gstreamer-vaapi
plugins so that these are not invoked by default
rather should only be tried through manual pipelines
when the user is sure about the caps matching.

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 gst/vaapi/gstvaapi.c      | 18 +++++++++---------
 gst/vaapi/gstvaapiparse.c |  4 ++--
 2 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/gst/vaapi/gstvaapi.c b/gst/vaapi/gstvaapi.c
index 3ad0ec0..f24c25d 100644
--- a/gst/vaapi/gstvaapi.c
+++ b/gst/vaapi/gstvaapi.c
@@ -54,34 +54,34 @@ static gboolean
 plugin_init (GstPlugin * plugin)
 {
   gst_element_register (plugin, "vaapidecode",
-      GST_RANK_PRIMARY + 1, GST_TYPE_VAAPIDECODE);
+      GST_RANK_PRIMARY - 1, GST_TYPE_VAAPIDECODE);
   gst_element_register (plugin, "vaapipostproc",
-      GST_RANK_PRIMARY, GST_TYPE_VAAPIPOSTPROC);
+      GST_RANK_PRIMARY - 1, GST_TYPE_VAAPIPOSTPROC);
   gst_element_register (plugin, "vaapisink",
-      GST_RANK_PRIMARY, GST_TYPE_VAAPISINK);
+      GST_RANK_PRIMARY - 1, GST_TYPE_VAAPISINK);
 #if USE_ENCODERS
   gst_element_register (plugin, "vaapiencode_h264",
-      GST_RANK_PRIMARY, GST_TYPE_VAAPIENCODE_H264);
+      GST_RANK_PRIMARY - 1, GST_TYPE_VAAPIENCODE_H264);
   gst_element_register (plugin, "vaapiencode_mpeg2",
-      GST_RANK_PRIMARY, GST_TYPE_VAAPIENCODE_MPEG2);
+      GST_RANK_PRIMARY - 1, GST_TYPE_VAAPIENCODE_MPEG2);
 #endif
 #if USE_JPEG_ENCODER
   gst_element_register (plugin, "vaapiencode_jpeg",
-      GST_RANK_PRIMARY, GST_TYPE_VAAPIENCODE_JPEG);
+      GST_RANK_PRIMARY - 1, GST_TYPE_VAAPIENCODE_JPEG);
 #endif
 #if USE_VP8_ENCODER
   gst_element_register (plugin, "vaapiencode_vp8",
-      GST_RANK_PRIMARY, GST_TYPE_VAAPIENCODE_VP8);
+      GST_RANK_PRIMARY - 1, GST_TYPE_VAAPIENCODE_VP8);
 #endif
 
 #if USE_H265_ENCODER
   gst_element_register (plugin, "vaapiencode_h265",
-      GST_RANK_PRIMARY, GST_TYPE_VAAPIENCODE_H265);
+      GST_RANK_PRIMARY - 1, GST_TYPE_VAAPIENCODE_H265);
 #endif
 
 #if GST_CHECK_VERSION(1,4,0)
   gst_element_register (plugin, "vaapidecodebin",
-      GST_RANK_PRIMARY + 2, GST_TYPE_VAAPI_DECODE_BIN);
+      GST_RANK_PRIMARY - 1, GST_TYPE_VAAPI_DECODE_BIN);
 #endif
   return TRUE;
 }
diff --git a/gst/vaapi/gstvaapiparse.c b/gst/vaapi/gstvaapiparse.c
index 06ac69c..9e9165f 100644
--- a/gst/vaapi/gstvaapiparse.c
+++ b/gst/vaapi/gstvaapiparse.c
@@ -39,11 +39,11 @@ plugin_init (GstPlugin * plugin)
   gboolean failure = FALSE;
 
   failure |= !gst_element_register (plugin, "vaapiparse_h264",
-      GST_RANK_PRIMARY + 2, GST_TYPE_H264_PARSE);
+      GST_RANK_PRIMARY - 1, GST_TYPE_H264_PARSE);
 
 #if GST_CHECK_VERSION(1,4,0)
   failure |= !gst_element_register (plugin, "vaapiparse_h265",
-      GST_RANK_PRIMARY + 2, GST_TYPE_H265_PARSE);
+      GST_RANK_PRIMARY - 1, GST_TYPE_H265_PARSE);
 #endif
 
   return !failure;
-- 
1.9.1

